From 3b6ad53b746c30427c10a7df5b2a238fe65cd238 Mon Sep 17 00:00:00 2001
Message-Id: <3b6ad53b746c30427c10a7df5b2a238fe65cd238.1413837856.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Pallavi <pallavi.g@intel.com>
Date: Thu, 30 May 2013 16:46:27 +0530
Subject: [PATCH 029/312] FOR_UPSTREAM [VPG]: drm/i915: Disable sprite planes

Disable the sprite planes as part of the display disable sequence
This change is specific to BYT-M/D.

Issue: APDEV-1157
Signed-off-by: Pallavi <pallavi.g@intel.com>
Change-Id: I9d3ecc14e98a4ab8a39ad121b4676d90340637e0
Reviewed-on: http://android.intel.com:8080/110788
Reviewed-by: cactus <cactus@intel.com>
Reviewed-by: Kumar, Shobhit <shobhit.kumar@intel.com>
Tested-by: Stimson, Dale B <dale.b.stimson@intel.com>
Reviewed-by: Yang, Fei <fei.yang@intel.com>
Reviewed-by: Wood, Brian J <brian.j.wood@intel.com>
Reviewed-by: Stimson, Dale B <dale.b.stimson@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
Signed-off-by: A.Sunil Kamath <sunil.kamath@intel.com>
Signed-off-by: Jon Bloomfield <jon.bloomfield@intel.com>
---
 drivers/gpu/drm/i915/i915_reg.h      |    1 +
 drivers/gpu/drm/i915/intel_display.c |   15 +++++++++++++++
 2 files changed, 16 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_reg.h b/drivers/gpu/drm/i915/i915_reg.h
index ae5f2f9..0994c16 100644
--- a/drivers/gpu/drm/i915/i915_reg.h
+++ b/drivers/gpu/drm/i915/i915_reg.h
@@ -4315,6 +4315,7 @@ enum punit_power_well {
 #define _SPBTILEOFF		(VLV_DISPLAY_BASE + 0x722a4)
 #define _SPBCONSTALPHA		(VLV_DISPLAY_BASE + 0x722a8)
 #define _SPBGAMC		(VLV_DISPLAY_BASE + 0x722f4)
+#define VLV_NUM_SPRITES		2
 
 #define SPCNTR(pipe, plane) _PIPE(pipe * 2 + plane, _SPACNTR, _SPBCNTR)
 #define SPLINOFF(pipe, plane) _PIPE(pipe * 2 + plane, _SPALINOFF, _SPBLINOFF)
diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 3979bf2..df5e4e3 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -2106,6 +2106,21 @@ static void intel_enable_pipe(struct intel_crtc *crtc)
 
 	I915_WRITE(reg, val | PIPECONF_ENABLE);
 	POSTING_READ(reg);
+
+	/* disable the sprite planes */
+	if (IS_VALLEYVIEW(dev_priv->dev)) {
+		int i;
+		for (i = 0; i < VLV_NUM_SPRITES; i++) {
+			val = I915_READ(SPCNTR(pipe, i));
+			if ((val & SP_ENABLE) == 0)
+				break;
+
+			I915_WRITE(SPCNTR(pipe, i), (val & ~SP_ENABLE));
+			/* Activate double buffered register update */
+			I915_WRITE(SPSURF(pipe, i), 0);
+			POSTING_READ(SPSURF(pipe, i));
+		}
+	}
 }
 
 /**
-- 
1.7.9.5

